pkg/server/server.go

```diff
@@ -701,10 +705,13 @@ func (s *Server) onConn(conn *clientConn) {
 	logutil.Logger(ctx).Debug("new connection", zap.String("remoteAddr", conn.bufReadConn.RemoteAddr().String()))
 
 	defer func() {
+		conn.decrementUserConnectionsCounter()
 		terror.Log(conn.Close())
 		logutil.Logger(ctx).Debug("connection closed")
 	}()
 
+	conn.incrementUserConnectionsCounter()
```

Comment:
this is the only invocation of `incrementUserConnectionsCounter`. Please also consider `handleChangeUser`

Reply:
This part of function is  compatible with mysql~

Reply:
OK, please add comment to explain that COM_CHANGE_USER will not be considered in MySQL

Reply:
Added comments at the head of increaseUserConnectionsCount().